hack/sync.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+#
+# Copyright 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source ./hack/common.sh
+
+# Please note that the validation webhooks are not currently deployed so take care when using a custom SSP CR
+export KUBEVIRT_SSP=${KUBEVIRT_SSP:-./config/samples/ssp_v1beta1_ssp.yaml}
+
+# The ssps.ssp.kubevirt.io CRD might not be present if this is the first sync so check before attempting to delete any actual objects
+if "${KUBECTL}" get crd/ssps.ssp.kubevirt.io; then
+    "${KUBECTL}" delete --ignore-not-found=true ssps -n kubevirt --all
+fi
+# The remaining CRDs should be present already from the k8s and KubeVirt installs so just ignore object not found failures
+"${KUBECTL}" delete --ignore-not-found=true deployment/ssp-operator -n kubevirt
+"${KUBECTL}" delete --ignore-not-found=true virtualmachineclusterinstancetype --all
+"${KUBECTL}" delete --ignore-not-found=true virtualmachineclusterpreference --all
+
+nodes=()
+for i in $(seq 1 "${KUBEVIRT_NUM_NODES}"); do
```

Comment:
https://github.com/search?q=org%3Akubevirt+KUBEVIRT_NUM_NODES&type=code

Reply:
Ah apologies I did think I was sourcing the individual common.sh files for each path, I've added these in now.

Reply:
Actually wait that's wrong I don't need to source the kubevirtci common.sh here, if it's set we will see it as an env variable otherwise we just treat it as a single host cluster and push the image there. I'll quickly test to demonstrate.

Reply:
Yeah this is working with the variable exported as it would be when used.

```
$ env | grep KUBEVIRT
KUBEVIRT_PROVIDER=k8s-1.24
KUBEVIRT_MEMORY=32768
KUBEVIRT_STORAGE=rook-ceph-default
$ export KUBEVIRT_NUM_NODES=2
$ make cluster-up && SSP_BUILD_RUNTIME=docker make cluster-sync
[..]
$ ./hack/kubevirtci.sh ssh node01 sudo docker images registry:5000/kubevirt/ssp-operator
[..]
selecting docker as container runtime
REPOSITORY                           TAG         IMAGE ID      CREATED        SIZE
registry:5000/kubevirt/ssp-operator  latest      be7bb2c87275  5 minutes ago  206 MB
$ ./hack/kubevirtci.sh ssh node02 sudo docker images registry:5000/kubevirt/ssp-operator
[..]
selecting docker as container runtime
REPOSITORY                           TAG         IMAGE ID      CREATED        SIZE
registry:5000/kubevirt/ssp-operator  latest      be7bb2c87275  5 minutes ago  206 MB
```

Reply:
Ok